apps/client/src/components/Location/Location.js

```diff
@@ -51,8 +51,16 @@ const Location = ({ topic }) => {
 
       // Reset the checker and answers when the address is changed
       if (answers && sessionAddress.id !== address.id) {
-        checkerContext.checker = null;
+        sessionContext.setSessionData([
+          slug,
+          {
+            finishedLocation: false,
+            finishedQuestions: false,
+          },
+        ]);
         answers = null;
+        setChecker(null);
+        initChecker();
```

Comment:
Can we make this into one function: `resetChecker();` because you're passing two function all the way down.

And in `apps/client/src/hoc/withChecker.js` we make a `resetChecker = () =>` function which only does:
```
setChecker(null);
initChecker();
```